cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -166,10 +208,103 @@ func (s *Supervisor) loadConfig(configFile string) error {
 	return nil
 }
 
-// TODO: Implement bootstrapping https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21071
-// nolint: unparam
 func (s *Supervisor) getBootstrapInfo() (err error) {
-	s.agentVersion = "1.0.0"
+	port, err := s.findRandomPort()
+	if err != nil {
+		return err
+	}
+
+	supervisorPort, err := s.findRandomPort()
+	if err != nil {
+		return err
+	}
+
+	var cfg bytes.Buffer
+
+	err = s.bootstrapTemplate.Execute(&cfg, map[string]any{
+		"EndpointPort":   port,
+		"InstanceUid":    s.instanceID.String(),
+		"SupervisorPort": supervisorPort,
+	})
+	if err != nil {
+		return err
+	}
+
+	s.writeEffectiveConfigToFile(cfg.String(), s.effectiveConfigFilePath)
+
+	srv := server.New(s.logger.Sugar())
+
+	done := make(chan struct{}, 1)
+	var connected atomic.Bool
+
+	err = srv.Start(newServerSettings(flattenedSettings{
+		endpoint: fmt.Sprintf("localhost:%d", supervisorPort),
+		onConnectingFunc: func(request *http.Request) {
+			connected.Store(true)
+
+		},
+		onMessageFunc: func(_ serverTypes.Connection, message *protobufs.AgentToServer) {
+			if message.AgentDescription != nil {
+				s.agentDescription = message.AgentDescription
+				identAttr := s.agentDescription.IdentifyingAttributes
+
+				for _, attr := range identAttr {
+					switch attr.Key {
+					case semconv.AttributeServiceInstanceID:
+						// TODO Consider this a critical error
```

Comment:
maybe worth looking at propagating this error now? Can we make the chan pass the error back to the main thread?

Reply:
Done, let me know what you think.